pkg/doc/verifiable/credential.go

```diff
@@ -421,9 +605,75 @@ func loadCredentialSchema(url string, client *http.Client) ([]byte, error) {
 	return gotBody, nil
 }
 
-// JSONBytes converts Verifiable Credential to JSON bytes
-func (vc *Credential) JSONBytes() ([]byte, error) {
-	rawCred := &rawCredential{
+// JWT serializes Credential to signed JWT.
+func (vc *Credential) JWT(signatureAlg JWTAlgorithm, privateKey interface{}, keyID string, minimizeVc bool) (string, error) { // nolint:lll
+	// currently jwt encoding supports only single subject (by the spec)
+	jwtClaims := &jwt.Claims{
+		Issuer:    vc.Issuer.ID,                   // iss
+		NotBefore: jwt.NewNumericDate(*vc.Issued), // nbf
+		ID:        vc.ID,                          // jti
+		Subject:   vc.getFirstSubjectID(),         // sub
+		IssuedAt:  jwt.NewNumericDate(*vc.Issued), // iat (not in spec, follow the interop project approach)
+	}
+	if vc.Expired != nil {
+		jwtClaims.Expiry = jwt.NewNumericDate(*vc.Expired) // exp
+	}
+
+	var raw *rawCredential
+	if minimizeVc {
+		vcCopy := *vc
+		vcCopy.Expired = nil
+		vcCopy.Issuer.ID = ""
+		vcCopy.Issued = nil
+		vcCopy.ID = ""
+		raw = vcCopy.raw()
+	} else {
+		raw = vc.raw()
+	}
+
+	credClaims := credentialClaims{
+		Claims:     jwtClaims,
+		Credential: raw,
+	}
+
+	key := jose.SigningKey{Algorithm: signatureAlg.Jose(), Key: privateKey}
+
+	var signerOpts = &jose.SignerOptions{}
+	signerOpts.WithType("JWT")
+	signerOpts.WithHeader("kid", keyID)
+
+	rsaSigner, err := jose.NewSigner(key, signerOpts)
+	if err != nil {
+		return "", fmt.Errorf("failed to create signer: %w", err)
+	}
+
+	// create an instance of Builder that uses the rsa signer
+	builder := jwt.Signed(rsaSigner)
+
+	builder = builder.Claims(credClaims)
+
+	// validate all ok, sign with the RSA key, and return a compact JWT
+	rawJWT, err := builder.CompactSerialize()
+	if err != nil {
+		return "", fmt.Errorf("failed to sign JWT")
```

Comment:
to be discussed, wrap err or not here ..

Reply:
Fixed it by error wrapping at #382